File: app/src/main/java/com/mirash/familiar/preferences/AppPreferences.java
package com.mirash.familiar.preferences;

import android.content.Context;
import android.content.SharedPreferences;

import com.mirash.familiar.Const;
import com.mirash.familiar.FamiliarApp;

/**
 * @author Mirash
 */

public class AppPreferences extends BasePreferences {
    private static AppPreferences instance;

    public AppPreferences(Context context) {
        super(context);
    }

    @Override
    protected SharedPreferences initPreferences(Context context) {
        return context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    public static AppPreferences getInstance() {
        if (instance == null) {
            instance = new AppPreferences(FamiliarApp.getInstance().getApplicationContext());
        }
        return instance;
    }

    public int getInvalidInputsCount() {
        return getInt(Const.KEY_INVALID_INPUT_COUNT, 0);
    }

    public void setInvalidInputsCount(int value) {
        putInt(Const.KEY_INVALID_INPUT_COUNT, value);
    }

    public void removeInvalidInputsCount() {
        remove(Const.KEY_INVALID_INPUT_COUNT);
    }
}

File: app/src/main/java/com/mirash/familiar/preferences/AppPreferences.kt
package com.mirash.familiar.preferences

import android.content.Context
import android.content.SharedPreferences
import com.mirash.familiar.FamiliarApp
import com.mirash.familiar.tool.KEY_INVALID_INPUT_COUNT
import com.mirash.familiar.tool.KEY_USER_ID

/**
 * @author Mirash
 */
object AppPreferences : BasePreferences(FamiliarApp.instance) {

    override fun initPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences(context.packageName, Context.MODE_PRIVATE)
    }

    fun getInvalidInputsCount(): Int = getInt(KEY_INVALID_INPUT_COUNT, 0)

    fun setInvalidInputsCount(value: Int) {
        putInt(KEY_INVALID_INPUT_COUNT, value)
    }

    fun removeInvalidInputsCount() {
        remove(KEY_INVALID_INPUT_COUNT)
    }

    fun getUserId(): Long = getLong(KEY_USER_ID, 0)

    fun setUserId(value: Long) {
        putLong(KEY_USER_ID, value)
    }
}
